src/code/modules/notifications.js

```diff
@@ -27,22 +27,30 @@ export default function notifications(state = initialState, action) {
       if (action.data) {
         let data = GuideProtocol.TutorDialog.fromJson(action.data);
         if (data.reason) {
-          console.log(`%c ITS Message Reason: ${data.reason.why || ""}`, `color: #f99a00`, action.data.reason);
+          console.log(`%c ITS Message Reason: ${data.reason.why || ""}`, `color: #f99a00`, data.reason);
         }
 
         if (state.messages[0] && state.messages[0].type === notificationType.NARRATIVE) {
           console.log(`%c Not showing ITS Message due to narrative`, `color: #f99a00`);
           return state;
         }
 
-        let trait = data.message.args ? data.message.args.trait : null;
         let currentText = state.messages.length > 0 ? state.messages[0].text + " " : "";
         let newMessage = {
           text: currentText + data.message.asString()
         };
+
+        let trait;
+        // check reason.trait first, then message.args.trait
```

Comment:
Is the idea that the ITS sends over information about the incorrect trait in different ways for different problem types?

Reply:
They haven't implemented `reason.trait` yet, and I'm not sure of their timeline to do so. The same info just so happens to be in `message.args.trait`, but it wasn't really designed for that (that's supposed to be for string substitution, I guess), so we shouldn't rely on it being there. So... there you have it.